src/forest.cc

```diff
@@ -428,12 +428,8 @@ void Forest::sampleNextGenealogy( bool recordEvents ) {
     // Don't implement a recombination if we are just here because rates changed
     dout << std::endl ;
     scrmdout << "Position: " << this->current_base() << ": Changing rates." << std::endl;
-    this->record_Recombevent_atNewGenealogy( 0 );
-    this->sampleNextBase();
-    this->record_Recombevent_b4_extension();
-    this->calcSegmentSumStats();
-    dout << "Next Position: " << this->next_base() << std::endl;
-    return;
+    //this->record_Recombevent_atNewGenealogy( 0 );
+    return 0; //enables the above line to be carried out at a higher level
```

Comment:
i was wondering why do you want to return 0 here? https://github.com/luntergroup/scrm_jz_stable_branch/blob/8690d3fd7c7dd2f935f7ac9e3dec6c793d6fc4e5/tests/algorithmtest/test_algorithm.cc#L55
also calls this function